### PR TITLE
Use new JWT Secret when signing/verifying Tokens

### LINKED ISSUE #1871: [auth] Add JWT Signing
We need to include JWT signing so our tokens can be verified with a public key.  Right now we only use a secret from the env.  We'll have to figure out how to generate, store, and pass the public/private keys into the container from staging/prod, similar to what we do now with certs for SSL and nginx.

### PR DESCRIPTION
**Pull request by joelazwar**
## Issue This PR Addresses
Fixes #1871 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Replace env variable SECRET with new JWT_SECRET for Supabase Integration

Satellite changes [here](https://github.com/Seneca-CDOT/satellite/pull/61)

## Steps to test the PR
1. `cp config/env.development .env`
2. `pnpm jest token.test.js`

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by DukeManh — APPROVED** [review]

**Review by tcvan0707 — DISMISSED** [review]

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -23,8 +23,12 @@ services:
       # Satellite authentication/authorization support
       - JWT_ISSUER
       - JWT_AUDIENCE
+<<<<<<< HEAD
```
Comment (menghif) [inline docker/docker-compose.yml:26]: This was left from the rebase

**Review by menghif — CHANGES_REQUESTED** [review]

File: src/api/sso/src/index.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const service = new Satellite({
     app.use(
       session({
         store: process.env.NODE_ENV === 'production' && new RedisStore({ client: Redis() }),
-        secret: process.env.SECRET || `telescope-has-many-secrets-${Date.now()}!`,
+        secret: process.env.JWT_SECRET || `telescope-has-many-secrets-${Date.now()}!`,
```
Comment (humphd) [inline src/api/sso/src/index.js:18]: Does this default secret need to be in any particular form?  `telescope-has-many-secrets-${Date.now()}!` is 41 chars btw.

**Comment by humphd** [comment]
This is blocked on https://github.com/Seneca-CDOT/satellite/pull/61#pullrequestreview-885321639.

File: docker/docker-compose.yml (hunk 1 above)
Comment (tcvan0707) [reply docker/docker-compose.yml]: Surprisingly, I didn't see this one when reviewing it

File: src/api/sso/src/index.js (hunk 2 above)
Comment (joelazwar) [reply src/api/sso/src/index.js:18]: https://github.com/Seneca-CDOT/telescope/blob/0723bb957f453ad6f80e690307b711d1b548209d/config/env.development#L279

at least 32 chars long

Comment (joelazwar) [reply src/api/sso/src/index.js:18]: the old secret

https://github.com/Seneca-CDOT/telescope/blob/0723bb957f453ad6f80e690307b711d1b548209d/config/env.development#L76

Comment (humphd) [reply src/api/sso/src/index.js:18]: OK, so this is fine.

We'll have to figure out what our solution is going to be in staging/production for these keys.

**Comment by humphd** [comment]
@joelazwar FYI, I just rebased this (on GitHub), so I could re-run the tests and see if that e2e failure will get fixed with @Kevan-Y's change.

**Comment by humphd** [comment]
This is failing due to the Users service being removed:

```
FetchError: invalid json response body at http://localhost/v1/users/33093c53dd reason: Unexpected token p in JSON at position 4
```

It shouldn't be trying to hit the users service, which we removed last night I thought.

**Comment by humphd** [comment]
I did some more work locally, and this is crashing waiting on the Satellite fix:

```
  if (!options || !options.secret) throw new Error('secret should be set');
                                   ^

Error: secret should be set
    at module.exports (/home/node/app/node_modules/express-jwt/lib/index.js:20:42)
    at isAuthenticated (/home/node/app/node_modules/@senecacdot/satellite/src/middleware.js:8:10)
    at Object.<anonymous> (/home/node/app/src/routes/register.js:17:3)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/node/app/src/routes/index.js:8:25)
```

**Comment by humphd** [comment]
This was fixed and merged in 6c2713c8e2d6a486b33279f4913d30ddd325d694

**Pull request closed without merging**